src/wp-includes/option.php

```diff
@@ -1133,6 +1140,65 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determines the appropriate autoload value for an option based on input.
+ *
+ *  This function checks the provided autoload value and returns a standardized
+ *  value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * If no valid autoload value is provided, the function will check the size of the given option value
+ * and compare it to the maximum allowed size. If it ls larger, it will return 'default-no', otherwise
+ * it will return 'default-yes'.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param string           $option   The name of the option.
+ * @param mixed            $value    The value of the option to check its autoload value.
+ * @param string|bool|null $autoload The autoload value to check.
+ *                                   Accepts 'yes'|true to enable or 'no'|false to disable, or
+ *                                   'default-yes' and 'default-no' for internal purposes.
+ *                                   Any other autoload value will be forced to either `default-yes`
+ *                                   or `default-no`.
+ * @return string Returns the original $autoload value if valid, or 'default-no' if the size exceeds
+ *                the maximum allowed size, or 'default-yes' otherwise.
+ */
+function determine_option_autoload_value( $option, $value, $autoload ) {
+
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+
+	}
+	// Serialize the value and check its size against the maximum allowed option size.
+	$serialized_value = maybe_serialize( $value );
```

Comment:
Don't think this is necessary either. The `determine_option_autoload_value()` is a low-level utility function for a very specific purpose. Imho it should be used only after the option value is serialized (as above), and the `$value` param should be a string.